include/deal.II/grid/grid_out.h

```diff
@@ -823,6 +828,7 @@ namespace GridOutFlags
         const bool         label_cell_index               = false,
         const bool         label_material_id              = false,
         const bool         label_subdomain_id             = false,
+        const bool         label_boundary_id              = false,
         const bool         draw_colorbar                  = false,
         const bool         draw_legend                    = false);
```

Comment:
The placement of the new function argument makes the change incompatible to what you had before. That's because if someone provided all of the boolean arguments before, the call is still correct but what used to be interpreted as `draw_colorbar` is now the `label_boundary_id` argument.

It's unfortunate, but the only way to do this in a backward compatible way is to put the new argument at the end.

(If we had thought this through some more, we'd have made the `label` kind of arguments an `enum` bitfield where one can select any number of bits, and just has to pass a single argument. But that's for a separate patch.

Reply:
Ooo good point. Even with the incompatibility changelog it could cause unexplained changes in behavior on the user's end. I'll put it at the bottom.

Would that make this no longer an incompatibility?

Reply:
No: Existing code only provided at most N arguments, so the N+1st argument is left at its default, which is `false`.

Reply:
Cool! I'll update the changelog then.